proposals/lip-0052.md

```diff
@@ -30,329 +31,331 @@ Therefore, providing a unified module to handle, transfer and modify NFTs is a n
+To allow cross-chain transfers of NFTs, we define a specific command which makes use of the [Interoperability module][lip-0045] and creates a [cross-chain message][lip-0049] with the relevant information. When sending NFTs cross-chain, it is crucial that every native chain can correctly escrow its native NFTs sent to a foreign chain. In this way, a native NFT can never be created by a foreign chain and sent across the ecosystem. When receiving non-native NFTs on a chain, users can query this NFT's native chain to make sure that the NFT is properly escrowed.
 
 #### Transfer To and From the Native Chain
 
-These specifications only allow NFTs to be transferred to and from their native chain. In particular, this means that a token created on chain A cannot be transferred directly from chain B to chain C. This is required to allow the native chain to maintain correctly escrowed NFTs.
+These specifications only allow NFTs to be transferred from or to their native chain. In particular, this means that NFT created on chain A cannot be transferred directly from chain B to chain C. This is required to allow the native chain to maintain escrowed NFTs correctly.
 
 ### Attributes
 
-Each NFT is stored with an attribute property. This property is a byte sequence that is not deserialized by the NFT module. Each custom module using an NFT collection should define schemas to serialize and deserialize the attribute property of NFTs of their collection.
+Each NFT is stored with an array of attributes specified by various modules, with each `attribute` property being a byte sequence that is not deserialized by the NFT module. Each custom module using NFTs should define schemas to serialize and deserialize their `attributes` property of NFTs.
 
-When an NFT is sent to another chain, the attributes property of the NFT can be modified according to specifications set on the receiving chain. For this reason, custom modules specifying an NFT collection must also implement the behavior to adopt when an NFT is returned with a modified attributes property. This custom behavior will compare the returned attributes with the ones stored with the escrowed NFT. If the returned NFT has an empty attribute, the native chain will restore the attributes as stored, this can be used to save on cross-chain messages size when returning non-modified NFTs to their native chains.
+When an NFT is sent to another chain, the `attributes` properties of the NFT can be modified according to specifications set on the receiving chain, i.e., in the processing done after the cross-chain NFT transfer. For this reason, custom modules specifying an NFT must also implement the behavior to adopt when an NFT is returned with a modified `attributes` properties. By default, an NFT returning back to its native chain will have the attributes as saved in the escrow account.
```

Comment:
Once the topic is completely finalized (see my comment in CCM execution), it would be nice to explicitly say what a module should do in order to specify the cross-chain attribute processing logic.

E.g. for my proposal add something like: "Once an NFT is received back at its native chain, any module can perform custom attribute processing according to the logic specified in `afterCrossChainCommandExecution`. In particular, a module might update all the attributes to the values received cross-chain, or keep the old values, or implement more complicated processing."

Reply:
Noted.